### ISSUE TITLE
AFNI dereferences files while accessing them

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#33]
thus being uncapable to open .BRIKs since .HEADs would be in another key directory:

```
$> afni 0back_SD_tbi-hc+tlrc.BRIK  
   
Thanks go to J Haxby for much encouragement

Initializing: X11.. Widgets...... Input files:ONLY ATLASES
** Couldn't open .../.git/annex/objects/fW/24/SHA256E-s2280000--b3b0....BRIK as session OR as dataset!

```

Workaround -- use ```AFNI_NOREALPATH=yes```

**Comment by hanke** [comment]
This is a symptom that we could see frequently, probably not every time coming with a nice workaround.

In my opinion this leads to a basic design decision: do we want people to work in annex'd directories?

If not, then we could use a "crawler"-like behavior for the generation and update of dataset handles. I.e. a sentinel job monitors a configured directory and reflects changes in the annex, but leaves the original directory untouched. Instead of a constant monitoring it could also be round-based and hand-triggered ("now I am ready for revision 2").

It get's more complicate on the consumption side of dataset handles. I obtain a handle `git annex get` its content, and get it served in an un-annexed directory, where its content is hard-linked into the actual annex. git annex could then monitor the "detached checkout" to identify modified/added/removed files. I have no idea how this would work in combination with meta-data based views of a dataset.

Needs more thinking...

**Comment by yarikoptic** [comment]
also do not forget about "direct mode".  For datasets managed exclusively with datalad, and thus git-annex, we might escape all the symlinks madness by simply using direct mode "installations". Side effect -- without "lean view" ( #25 ) there would be empty (0 size) files present.

**Comment by hanke** [comment]
Yeah, direct mode is there. And maybe that is what we should think about as a default mode of operation. Benjamin brought up that the git annex setup as a whole is more complicated on windows. Same goes for mapping possible workflows fro. Linux to windows. Maybe we should constrain our thinking to what is possible on windows right away? The thought alone pains me...

**Comment by bpoldrack** [comment]
Let me figure things out before considering such painful thoughts. The
problems I experience on Windows give me a fast track to a deeper insight.
So that's the upside for now. As far as I can see for now, the problem is:
Using direct mode leads to the question Michael brought up:

In my opinion this leads to a basic design decision: do we want people to
> work in annex'd directories?


I have little experience with the targeted kind of users but I'm afraid
forcing users to do or not to do it this way may be lead to low acceptance
and lots of complains. Am I wrong here?


2015-01-22 16:32 GMT+01:00 Michael Hanke <notifications@github.com>:

**Comment by yarikoptic** [comment]
IMHO we should just have a documentation section outlining such gotchas and workarounds (env var for AFNI, direct mode resolution), thus marking as documentation and assigning to @hanke ;)

**Comment by mih** [comment]
When #2357 is merged, move to https://github.com/datalad/datalad-neuroimaging
